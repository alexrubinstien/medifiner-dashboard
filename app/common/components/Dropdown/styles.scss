@import './app/common/styles/utils/_vars.scss';

.dropdown {
  position: relative;
  margin: -16px -20px -10px;

  &--wide-content {
    position: static;
  }

  &__open-container {
    padding: 16px 20px 10px;
    border-radius: 3px 3px 0 0;
    transition: all .25s ease;

    &--is-open {
      background: $primary-color;
    }
  }

  &__label {
    margin: 0 0 10px;
    color: $text-color;
    font-size: 12px;
    font-weight: 700;
    line-height: 20px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }

  &__toggle {
    width: 100%;
    padding: 0 0 8px;
    background: transparent;
    border: 0;
    border-bottom: 2px solid $border-color;
    border-radius: 0;
    color: $primary-color;
    font-size: 16px;
    font-weight: 500;
    line-height: 26px;
    letter-spacing: 1px;
    text-align: left;
    transition: all .25s ease;

    &--placeholder {
      color: $text-light-color;
    }
  }

  &__toggle-icon {
    float: right;
    color: $primary-color;
    font-size: 8px;
  }

  &__open-container--is-open &__label,
  &__open-container--is-open &__toggle {
    color: #fff;
  }

  &__list-container {
    position: absolute;
    right: 0;
    left: 0;
    z-index: 1001;
    background: transparent;
    border-radius: 0 0 3px 3px;
    transition: all .25s ease;


    &--is-open {
      background: $primary-color;
    }
  }

  &--wide-content &__list-container:before {
    content: '';
    position: absolute;
    top: 0;
    bottom: 0;
    left: 20%;
    width: 2px;
    background: $primary-color-hover;
    opacity: .3;
  }

  &__categories-container {
    float: left;
    width: 20%;
    padding: 22px 30px;
  }

  &__categories {
    margin: 0;
    padding: 0;
    list-style: none;
  }

  &__category {
    background: transparent;
    border: 0;
    border-radius: 0;
    color: #fff;
    font-size: 16px;
    font-weight: 700;
    line-height: 26px;
  }

  &__columns-container {
    float: left;
    overflow-y: auto;
    width: 80%;
    max-height: 506px;
    padding: 0;
  }

  &__items-list-column {
    float: left;
    width: 25%;
  }

  &__items-list {
    margin: 0;
    padding: 22px 26px;
    list-style: none;
  }

  &__item-button {
    width: 100%;
    padding: 2px 20px 4px;
    background: transparent;
    border: 0;
    border-radius: 18.5px;
    color: $background-3;
    font-size: 16px;
    font-weight: 500;
    line-height: 26px;
    text-align: left;
    transition: all .25s ease;

    &--active,
    &--active:hover {
      background: $primary-color-hover;
    }

    &:hover:not(&--active) {
      background: rgba(19, 139, 219, .3);
    }
  }
}
