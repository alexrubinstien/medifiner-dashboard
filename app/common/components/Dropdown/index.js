import React, { Component } from 'react';
import PropTypes from 'prop-types';

import AnimateHeight from 'react-animate-height';
import classNames from 'classnames';

import './styles.scss';

class Dropdown extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showDropdown: false,
    };

    this.onDocumentClick = this.onDocumentClick.bind(this);
    this.toggleDropdown = this.toggleDropdown.bind(this);

    this.containerRef = React.createRef();
  }

  componentWillMount() {
    if (typeof document !== 'undefined') {
      document.addEventListener('mousedown', this.onDocumentClick, false);
    }
  }

  componentWillUnmount() {
    if (typeof document !== 'undefined') {
      document.removeEventListener('mousedown', this.onDocumentClick, false);
    }
  }

  onDocumentClick(e) {
    if (!this.containerRef.current.contains(e.target)) {
      this.setState({
        showDropdown: false,
      });
    }
  }

  onItemClick(value) {
    this.setState({
      showDropdown: false,
    }, () => {
      this.props.onItemClick(value);
    });
  }

  onCategoryClick(value) {
    this.setState({
      showDropdown: false,
    }, () => {
      this.props.onCategoryClick(value);
    });
  }

  toggleDropdown() {
    this.setState({
      showDropdown: !this.state.showDropdown,
    });
  }

  renderItemsColumns(activeOptionCategory) {
    const { activeValue } = this.props;
    const itemsPerColumn = Math.ceil(activeOptionCategory.items.length / 4);

    return Array(4).fill(undefined).map((_, columnIndex) => (
      <div
        key={columnIndex}
        className="dropdown__items-list-column"
      >
        <ul className="dropdown__items-list">
          {activeOptionCategory.items.slice(itemsPerColumn * columnIndex, (itemsPerColumn * (columnIndex + 1))).map((item, itemIndex) => (
            <li
              key={itemIndex}
              className="dropdown__item"
            >
              <button
                type="button"
                className={classNames('dropdown__item-button', { 'dropdown__item-button--active': item.name === activeValue })}
                onClick={() => this.onItemClick(item.name)}
              >
                {item.title}
              </button>
            </li>
          ))}
        </ul>
      </div>
    ));
  }

  renderSingleCategories() {
    const {
      activeValue,
      options,
    } = this.props;

    return (
      <ul className="dropdown__items-list">
        {options.map(option => (
          <li
            key={option.category.name}
            className="dropdown__item"
          >
            <button
              type="button"
              className={classNames('dropdown__item-button', { 'dropdown__item-button--active': option.category.name === activeValue })}
              onClick={() => this.onItemClick(option.category.name)}
            >
              {option.category.title}
            </button>
          </li>
        ))}
      </ul>
    );
  }

  render() {
    const { showDropdown } = this.state;
    const {
      activeCategory,
      activeValue,
      label,
      options,
      placeholder,
    } = this.props;
    const activeOptionCategory = options.find(option => option.category.name === activeCategory);
    let activeName = '';
    if (activeOptionCategory && activeOptionCategory.items && activeValue) {
      activeName = activeOptionCategory.items.find(item => item.name === activeValue).title;
    } else if (activeOptionCategory && activeOptionCategory.category) {
      activeName = activeOptionCategory.category.title;
    }

    return (
      <div
        className={classNames('dropdown', { 'dropdown--wide-content': activeOptionCategory && !!activeOptionCategory.items })}
        ref={this.containerRef}
      >
        <div className={classNames('dropdown__open-container', { 'dropdown__open-container--is-open': showDropdown })}>
          <p className="dropdown__label">
            {label}
          </p>
          <button
            type="button"
            className={classNames('dropdown__toggle', { 'dropdown__toggle--placeholder': !activeName && placeholder })}
            onClick={this.toggleDropdown}
          >
            {activeName || placeholder}
            <span
              className={classNames('dropdown__toggle-icon', {
                'icon-arrow-down': !showDropdown,
                'icon-arrow-up': showDropdown,
              })}
            />
          </button>
        </div>
        <div className={classNames('dropdown__list-container', { 'dropdown__list-container--is-open': showDropdown })}>
          <AnimateHeight
            duration={250}
            height={showDropdown ? 'auto' : 0}
          >
            {(activeOptionCategory && activeOptionCategory.items) ? (
              <div className="clearfix">
                <div className="dropdown__categories-container">
                  <ul className="dropdown__categories">
                    {options.map(option => (
                      <li
                        key={option.category.name}
                        className="dropdown__category-item"
                      >
                        <button
                          type="button"
                          className="dropdown__category"
                          onClick={() => this.onCategoryClick(option.category.name)}
                        >
                          {option.category.title}
                        </button>
                      </li>
                    ))}
                  </ul>
                </div>
                <div className="dropdown__columns-container cleafix">
                  {this.renderItemsColumns(activeOptionCategory)}
                </div>
              </div>
            ) : (
              <div className="clearfix">
                {this.renderSingleCategories()}
              </div>
            )}
          </AnimateHeight>
        </div>
      </div>
    );
  }
}

Dropdown.propTypes = {
  activeCategory: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  activeValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  label: PropTypes.string.isRequired,
  onCategoryClick: PropTypes.func,
  onItemClick: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    category: PropTypes.shape({
      name: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
      title: PropTypes.string,
    }),
    items: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
      title: PropTypes.string,
    })),
  })).isRequired,
  placeholder: PropTypes.string,
};

Dropdown.defaultProps = {
  activeCategory: null,
  activeValue: null,
  onCategoryClick: () => {},
  placeholder: null,
};

export default Dropdown;
