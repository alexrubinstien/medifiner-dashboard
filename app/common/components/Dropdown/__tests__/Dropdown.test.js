/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Dropdown from '../index.js';

describe('With Snapshot Testing', () => {
  it('Dropdown component with single list', () => {
    const component = renderer.create(
      <Dropdown
        activeCategory="oseltamivir"
        activeValue="oseltamivir"
        label="Average Supply Levels of"
        onItemClick={() => {}}
        options={[{
          category: {
            name: 'zanamivir',
            title: 'Zanamivir',
          },
        }, {
          category: {
            name: 'oseltamivir',
            title: 'Oseltamivir',
          },
        }, {
          category: {
            name: 'simvastatin',
            title: 'Simvastatin',
          },
        }, {
          category: {
            name: 'all antiviral drugs',
            title: 'All Antiviral Drugs',
          },
        }]}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Dropdown component with placeholder', () => {
    const component = renderer.create(
      <Dropdown
        placeholder="Select Medication"
        activeCategory=""
        activeValue=""
        label="Average Supply Levels of"
        onItemClick={() => {}}
        options={[{
          category: {
            name: 'zanamivir',
            title: 'Zanamivir',
          },
        }, {
          category: {
            name: 'oseltamivir',
            title: 'Oseltamivir',
          },
        }, {
          category: {
            name: 'simvastatin',
            title: 'Simvastatin',
          },
        }, {
          category: {
            name: 'all antiviral drugs',
            title: 'All Antiviral Drugs',
          },
        }]}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Dropdown component with multiple categories', () => {
    const component = renderer.create(
      <Dropdown
        activeCategory="usa"
        activeValue="alaska"
        label="Show in"
        onItemClick={() => {}}
        options={[{
          category: {
            name: 'usa',
            title: 'Uniated States',
          },
          items: [{
            name: 'alabama',
            title: 'Alabama',
          }, {
            name: 'alaska',
            title: 'Alaska',
          }, {
            name: 'arizona',
            title: 'Arizona',
          }, {
            name: 'arkansas',
            title: 'Arkansas',
          }, {
            name: 'california',
            title: 'California',
          }, {
            name: 'colorado',
            title: 'Colorado',
          }, {
            name: 'connecticut',
            title: 'Connecticut',
          }, {
            name: 'delaware',
            title: 'Delaware',
          }, {
            name: 'florida',
            title: 'Florida',
          }, {
            name: 'georgia',
            title: 'Georgia',
          }, {
            name: 'hawaii',
            title: 'Hawaii',
          }, {
            name: 'idaho',
            title: 'Idaho',
          }, {
            name: 'illinois',
            title: 'Illinois',
          }, {
            name: 'indiana',
            title: 'Indiana',
          }, {
            name: 'iowa',
            title: 'Iowa',
          }, {
            name: 'kansas',
            title: 'Kansas',
          }, {
            name: 'kentucky',
            title: 'Kentucky',
          }, {
            name: 'louisiana',
            title: 'Louisiana',
          }, {
            name: 'maine',
            title: 'Maine',
          }, {
            name: 'maryland',
            title: 'Maryland',
          }, {
            name: 'massachusetts',
            title: 'Massachusetts',
          }, {
            name: 'michigan',
            title: 'Michigan',
          }, {
            name: 'minnesota',
            title: 'Minnesota',
          }, {
            name: 'mississippi',
            title: 'Mississippi',
          }, {
            name: 'missouri',
            title: 'Missouri',
          }, {
            name: 'montana',
            title: 'Montana',
          }, {
            name: 'nebraska',
            title: 'Nebraska',
          }, {
            name: 'nevada',
            title: 'Nevada',
          }, {
            name: 'new hampshire',
            title: 'New Hampshire',
          }, {
            name: 'new jersey',
            title: 'New Jersey',
          }, {
            name: 'new mexico',
            title: 'New Mexico',
          }, {
            name: 'new york',
            title: 'New York',
          }, {
            name: 'north carolina',
            title: 'North Carolina',
          }, {
            name: 'north dakota',
            title: 'North Dakota',
          }, {
            name: 'ohio',
            title: 'Ohio',
          }, {
            name: 'oklahoma',
            title: 'Oklahoma',
          }, {
            name: 'oregon',
            title: 'Oregon',
          }, {
            name: 'pennsylvania',
            title: 'Pennsylvania',
          }, {
            name: 'rhode island',
            title: 'Rhode Island',
          }, {
            name: 'south carolina',
            title: 'South Carolina',
          }, {
            name: 'south dakota',
            title: 'South Dakota',
          }, {
            name: 'tennessee',
            title: 'Tennessee',
          }, {
            name: 'texas',
            title: 'Texas',
          }, {
            name: 'utah',
            title: 'Utah',
          }, {
            name: 'vermont',
            title: 'Vermont',
          }, {
            name: 'virginia',
            title: 'Virginia',
          }, {
            name: 'washington',
            title: 'Washington',
          }, {
            name: 'washington dc',
            title: 'Washington DC',
          }, {
            name: 'west virginia',
            title: 'West Virginia',
          }, {
            name: 'wisconsin',
            title: 'Wisconsin',
          }, {
            name: 'wyoming',
            title: 'Wyoming',
          }],
        }]}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
