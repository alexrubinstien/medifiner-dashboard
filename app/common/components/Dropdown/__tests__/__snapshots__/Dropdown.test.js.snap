// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Dropdown component with multiple categories 1`] = `
<div
  className="dropdown dropdown--wide-content"
>
  <div
    className="dropdown__open-container"
  >
    <p
      className="dropdown__label"
    >
      Show in
    </p>
    <button
      className="dropdown__toggle"
      onClick={[Function]}
      type="button"
    >
      Alaska
      <span
        className="dropdown__toggle-icon icon-arrow-down"
      />
    </button>
  </div>
  <div
    className="dropdown__list-container"
  >
    <div
      aria-hidden={true}
      className="rah-static rah-static--height-zero"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="clearfix"
        >
          <div
            className="dropdown__categories-container"
          >
            <ul
              className="dropdown__categories"
            >
              <li
                className="dropdown__category-item"
              >
                <button
                  className="dropdown__category"
                  onClick={[Function]}
                  type="button"
                >
                  Uniated States
                </button>
              </li>
            </ul>
          </div>
          <div
            className="dropdown__columns-container cleafix"
          >
            <div
              className="dropdown__items-list-column"
            >
              <ul
                className="dropdown__items-list"
              >
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Alabama
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button dropdown__item-button--active"
                    onClick={[Function]}
                    type="button"
                  >
                    Alaska
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Arizona
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Arkansas
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    California
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Colorado
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Connecticut
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Delaware
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Florida
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Georgia
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Hawaii
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Idaho
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Illinois
                  </button>
                </li>
              </ul>
            </div>
            <div
              className="dropdown__items-list-column"
            >
              <ul
                className="dropdown__items-list"
              >
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Indiana
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Iowa
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Kansas
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Kentucky
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Louisiana
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Maine
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Maryland
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Massachusetts
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Michigan
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Minnesota
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Mississippi
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Missouri
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Montana
                  </button>
                </li>
              </ul>
            </div>
            <div
              className="dropdown__items-list-column"
            >
              <ul
                className="dropdown__items-list"
              >
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Nebraska
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Nevada
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    New Hampshire
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    New Jersey
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    New Mexico
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    New York
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    North Carolina
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    North Dakota
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Ohio
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Oklahoma
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Oregon
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Pennsylvania
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Rhode Island
                  </button>
                </li>
              </ul>
            </div>
            <div
              className="dropdown__items-list-column"
            >
              <ul
                className="dropdown__items-list"
              >
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    South Carolina
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    South Dakota
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Tennessee
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Texas
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Utah
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Vermont
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Virginia
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Washington
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Washington DC
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    West Virginia
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Wisconsin
                  </button>
                </li>
                <li
                  className="dropdown__item"
                >
                  <button
                    className="dropdown__item-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Wyoming
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing Dropdown component with placeholder 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown__open-container"
  >
    <p
      className="dropdown__label"
    >
      Average Supply Levels of
    </p>
    <button
      className="dropdown__toggle dropdown__toggle--placeholder"
      onClick={[Function]}
      type="button"
    >
      Select Medication
      <span
        className="dropdown__toggle-icon icon-arrow-down"
      />
    </button>
  </div>
  <div
    className="dropdown__list-container"
  >
    <div
      aria-hidden={true}
      className="rah-static rah-static--height-zero"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="clearfix"
        >
          <ul
            className="dropdown__items-list"
          >
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button"
                onClick={[Function]}
                type="button"
              >
                Zanamivir
              </button>
            </li>
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button"
                onClick={[Function]}
                type="button"
              >
                Oseltamivir
              </button>
            </li>
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button"
                onClick={[Function]}
                type="button"
              >
                Simvastatin
              </button>
            </li>
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button"
                onClick={[Function]}
                type="button"
              >
                All Antiviral Drugs
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing Dropdown component with single list 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown__open-container"
  >
    <p
      className="dropdown__label"
    >
      Average Supply Levels of
    </p>
    <button
      className="dropdown__toggle"
      onClick={[Function]}
      type="button"
    >
      Oseltamivir
      <span
        className="dropdown__toggle-icon icon-arrow-down"
      />
    </button>
  </div>
  <div
    className="dropdown__list-container"
  >
    <div
      aria-hidden={true}
      className="rah-static rah-static--height-zero"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="clearfix"
        >
          <ul
            className="dropdown__items-list"
          >
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button"
                onClick={[Function]}
                type="button"
              >
                Zanamivir
              </button>
            </li>
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button dropdown__item-button--active"
                onClick={[Function]}
                type="button"
              >
                Oseltamivir
              </button>
            </li>
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button"
                onClick={[Function]}
                type="button"
              >
                Simvastatin
              </button>
            </li>
            <li
              className="dropdown__item"
            >
              <button
                className="dropdown__item-button"
                onClick={[Function]}
                type="button"
              >
                All Antiviral Drugs
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
